src/BlockTypesController.php

```diff
@@ -187,8 +187,14 @@ protected function get_block_types() {
 
 		if ( Package::feature()->is_experimental_build() ) {
 			$block_types[] = 'SingleProduct';
-			$block_types[] = 'MiniCart';
-			$block_types[] = 'MiniCartContents';
+
+			/**
+			 * Mini Cart blocks shouldn't be available in the Blocks Editor.
+			 */
+			if ( 'post.php' !== $pagenow ) {
+				$block_types[] = 'MiniCart';
+				$block_types[] = 'MiniCartContents';
+			}
```

Comment:
Currently, this check does a bit different from the original issue requirement. We're excluding the Mini Cart block from the Blocks Editor, while #5576 wants to **only allow adding the Mini Cart block inside the Site Editor and Widgets**. In the future, we may have other new editors that we may not want the Mini Cart to be available there.

The following code does exactly what #5576 requires:

```suggestion
			/**
			 * Mini Cart blocks should be available in Site Editor and Widgets only.
			 */
			if (
				'widgets.php' === $pagenow ||
				'site-editor.php' === $pagenow ||
				! empty( $_GET['page'] ) && 'gutenberg-edit-site' === $_GET['page'] // phpcs:ignore WordPress.Security.NonceVerification
			) {
				$block_types[] = 'MiniCart';
				$block_types[] = 'MiniCartContents';
			}
```